crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
> Do You suggest that the new ex_data already have some values?

Yes, because CRYPTO_new_ex_data() is called in the case of SSL_dup() for instance.

The destination values are not overwritten; it is under control of the dup function to assign a proper value to the **from_d** pointer to be assigned to the destination. The possibilities are:

1. Don't care: the ex_data value may not be an allocated pointer (i.e. a counter)
2. Copy the pointer: up function may just up-ref the source ex_data, and free the original ex_data (to avoid the leak)
3. Keep the new'd ex_data: update the **from_d** pointer to point to the new'd data, update ex_data contents with data from the source

ex_data use is non-trivial, it is way too easy to make a mistake.